controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -792,6 +827,7 @@ func (r *BareMetalHostReconciler) actionManageReady(prov provisioner.Provisioner
 		}
 		if dirty {
 			info.log.Info("updating host provisioning settings")
+			return actionUpdate{}
```

Comment:
Because I need different return actionResult to distinguish whether we need to go back to the `Preparing` state.

```go
func (hsm *hostStateMachine) handleReady(info *reconcileInfo) actionResult {
	if hsm.Host.Spec.ExternallyProvisioned {
		hsm.NextState = metal3v1alpha1.StateExternallyProvisioned
		return actionComplete{}
	}

	// ErrorCount is cleared when appropriate inside actionManageReady
	actResult := hsm.Reconciler.actionManageReady(hsm.Provisioner, info)
	if _, update := actResult.(actionUpdate); update {
		hsm.NextState = metal3v1alpha1.StatePreparing
	} else if _, complete := actResult.(actionComplete); complete {
		hsm.NextState = metal3v1alpha1.StateProvisioning
	}
	return actResult
}
```